### PR TITLE
Add enqueue functions

### PR DESCRIPTION
**Pull request by DevPabloGarcia**
### :goal_net: What's the goal?
In order to let the developer enqueue response, we replace old "provideDispatcher" function by two new functions

enqueue(mock: Mock)
enqueue(mocks: List<Mock)

This way the user can add responses in different parts of the code.

### :construction: How do we do it?
* Remove old function provideDispatcher
* Create a mutable list to store reponses
* Create functions to fill the mutable list with responses.

### :blue_book: Documentation changes?
- [ ] No docs to update nor create

### :test_tube: How can I test this?
- [ ] No tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,8 +59,8 @@ class App : Application() {
         ...
         if (BuildConfig.DEFAULT_ENVIRONMENT == Environment.DEMO) {
             CoroutineScope(Dispatchers.IO).launch {
-                mockHelper.startServer()
-                mockHelper.provideDispatcher(getUserMocksUseCase()) // This use case provide a List<Mock>
+                mockHelper.setup()
```
Comment (dagonco) [inline README.md:62]: Minor: I saw in the code `setUp` instead `setup`.

File: app/build.gradle [hunk 2]
```diff
@@ -12,8 +12,8 @@ android {
         applicationId "com.telefonica.mocks"
         minSdk 21
         targetSdk 32
-        versionCode 1
-        versionName "1.0"
+        versionCode 2
```
Comment (dagonco) [inline app/build.gradle:15]: I think that shouldnt be needed 🧹

File: mock/src/main/java/com/telefonica/mock/MockApiClient.kt [hunk 3]
```diff
@@ -11,14 +11,31 @@ import java.util.concurrent.TimeUnit
...
 ) {
 
     private val mockMap: MutableMap<String, List<Mock>> = mutableMapOf()
 
+    private val enqueuedAnswers: MutableList<Mock> = mutableListOf()
+
+    private val dispatcher = object : Dispatcher() {
+        override fun dispatch(request: RecordedRequest): MockResponse {
+            val mockFiles: List<Mock> = enqueuedAnswers
+                .filter { mock -> request.method == mock.method?.value }
```
Comment (dagonco) [inline mock/src/main/java/com/telefonica/mock/MockApiClient.kt:25]: Just curious, why method could be null?

**Review by dagonco — APPROVED** [review]
Great, nice job!

File: app/build.gradle (hunk 2 above)
Comment (DevPabloGarcia) [reply app/build.gradle:15]: Done!

File: README.md (hunk 1 above)
Comment (DevPabloGarcia) [reply README.md:62]: Done!

File: mock/src/main/java/com/telefonica/mock/MockApiClient.kt (hunk 3 above)
Comment (DevPabloGarcia) [reply mock/src/main/java/com/telefonica/mock/MockApiClient.kt:25]: When we developed this part, it made sense, maybe we could set it to not null and not leave it as the default.

**Review by dagonco — APPROVED** [review]

File: mock/src/main/java/com/telefonica/mock/MockApiClient.kt [hunk 4]
```diff
@@ -11,14 +11,31 @@ import java.util.concurrent.TimeUnit
...
     private val mockWebServer: MockWebServer,
 ) {
 
     private val mockMap: MutableMap<String, List<Mock>> = mutableMapOf()
 
+    private val enqueuedAnswers: MutableList<Mock> = mutableListOf()
+
+    private val dispatcher = object : Dispatcher() {
+        override fun dispatch(request: RecordedRequest): MockResponse {
+            val mockFiles: List<Mock> = enqueuedAnswers
```
Comment (yamal-alm) [inline mock/src/main/java/com/telefonica/mock/MockApiClient.kt:24]: how will this code be executed for the mocked responses enqueued after invoking `setUp`?

Comment (yamal-alm) [reply mock/src/main/java/com/telefonica/mock/MockApiClient.kt:24]: Oh, I see, this `dispatch` callback is executed on each request 👍

**Review by yamal-alm — APPROVED** [review]
looks good! 👌

**Pull request merged**